lib/util.ex

```diff
@@ -67,27 +73,33 @@ defmodule Util do
       Authorization: "Bearer " <> auth
     ]
 
-    {:ok, resp} = HTTPoison.post("https://api.start.gg/gql/alpha", JSON.encode!(body), headers)
-
-    resp.body
-    |> JSON.decode!()
-    |> Map.get("data")
-    |> Map.get("tournaments")
-    |> Map.get("nodes")
-    # Finesse data into our own non-start.gg format.
-    |> Enum.map(fn sgg_data ->
-      %{
-        "external_id" => "smashgg-" <> Integer.to_string(sgg_data["id"]),
-        "name" => sgg_data["name"],
-        "location" => %{
-          "lat" => sgg_data["lat"],
-          "lng" => sgg_data["lng"]
-        },
-        "start_time" => sgg_data["startAt"],
-        "end_time" => sgg_data["endAt"],
-        "updated_at" => sgg_data["updatedAt"],
-        "url" => "https://start.gg/" <> (sgg_data["slug"])
-      }
-    end)
+    case HTTPoison.post("https://api.start.gg/gql/alpha", JSON.encode!(body), headers) do
+      {:ok, resp} ->
+        case JSON.decode(resp.body) do
+          {:ok, response = %{"success" => false}} ->
+            IO.puts(Map.get(response, "message"))
+          {:ok, response} ->
```

Comment:
Just for completeness's sake, let's double-check that the data has `success: true` by matching for it here. If for some reason the data we get back does not have a `success` property, it will be more helpful to crash with an "unmatched data" error here, rather than a nil map error coming from further in.

For that matter, we could _almost_ go ahead and check that the data has the `data: { tournaments: { nodes: {...}}}` structure we expect by matching for it in this case statement, if it didn't require a triply-nested map... This might be worth coming back to later.

Reply:
Yeah I actually expected it to have an opposite `success: true` indicating a successful hit, but it doesn't and instead returns the `data: { tournaments: { nodes: {...}}}` shape. We could match on the triple nested shape here though, up to you!